query-engine/connectors/sql-query-connector/src/database/operations/write.rs

```diff
@@ -20,12 +20,13 @@ async fn generate_id(
     conn: &dyn QueryExt,
     primary_key: &FieldSelection,
     trace_id: Option<String>,
+    args: &WriteArgs,
 ) -> crate::Result<Option<SelectionResult>> {
     // Go through all the values and generate a select statement with the correct MySQL function
     let (pk_select, need_select) = primary_key
         .selections()
         .filter_map(|field| match field {
-            SelectedField::Scalar(x) if x.default_value.is_some() => x
+            SelectedField::Scalar(x) if x.default_value.is_some() && !args.has_arg_for(x.name()) => x
```

Comment:
`WriteArgs` contain the data source field name, which is not the `x.name()`, but the `x.db_name()`. Fairly sure that if you add a `@map` to the uuid field your test will fail.

Reply:
Fair enough